app/src/main/java/com/example/bucketbuds/ProfileFragment.java

```diff
@@ -1,65 +1,295 @@
+        // Create alert dialog builder
+        AlertDialog.Builder alertDialogBuilder = new AlertDialog.Builder(getContext());
+        alertDialogBuilder.setView(messageView);
+        // Create alert dialog
+        final AlertDialog alertDialog = alertDialogBuilder.create();
+        EditText etUsername = messageView.findViewById(R.id.etUsernameDE);
+        EditText etBio = messageView.findViewById(R.id.etBioDE);
+        EditText etFirstName = messageView.findViewById(R.id.etFirstNameDE);
+        EditText etLastName = messageView.findViewById(R.id.etLastNameDE);
+        EditText etEmail = messageView.findViewById(R.id.etEmailDE);
+        Button btnSave = messageView.findViewById(R.id.btnSaveDE);
+        Button btnCancel = messageView.findViewById(R.id.btnCancelDE);
+
+        etBio.setText(user.getBio(), TextView.BufferType.EDITABLE);
+        etUsername.setText(user.getUsername(), TextView.BufferType.EDITABLE);
+        etFirstName.setText(user.getFirstName(), TextView.BufferType.EDITABLE);
+        etLastName.setText(user.getLastName(), TextView.BufferType.EDITABLE);
+        etEmail.setText(user.getEmail(), TextView.BufferType.EDITABLE);
+
+        // Configure dialog buttons
+        btnSave.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+
+                String username = etUsername.getText().toString();
+                String bio = etBio.getText().toString();
+                String first = etFirstName.getText().toString();
+                String last = etLastName.getText().toString();
+                String email = etEmail.getText().toString();
+
+                if (username.isEmpty() || first.isEmpty() || last.isEmpty() || email.isEmpty()) {
+                    Toast.makeText(getContext(), "Name, username, and email are required", Toast.LENGTH_SHORT);
+                } else {
+                    alertDialog.cancel();
+                    user.setUsername(username);
+                    user.setBio(bio);
+                    user.setEmail(email);
+                    user.setFirstName(first);
+                    user.setLastName(last);
+                    user.saveInBackground(new SaveCallback() {
+                        @Override
+                        public void done(ParseException e) {
+                            if (e == null) {
+                                alertDialog.dismiss();
+                                tvBio.setText(bio);
+                                tvUsername.setText("@"+username);
+                                tvFirst.setText(first);
+                                tvLast.setText(last);
+                                alertDialog.dismiss();
+                            } else {
+                                Toast.makeText(getContext(), "Failed to save information", Toast.LENGTH_SHORT);
+                            }
+                        }
+                    });
+                }
+            }
+        });
```

Comment:
This could be split out to helper methods, or at least to variables to avoid nesting.
Nesting is when you have brackets inside brackets inside brackets like;
```
if(...) {
  while() {
    new X() {
      public void Y() {
        ...
     }
   } 
   for() {
   ....
   }
}
```
Nesting too much makes the code very hard to read.
Here, for example, you could do something like:
```
SaveCallback userSaveCb = new SaveCallback() {
  @Override 
  public void done(ParseException e) {
  ...
};
btnSave.setOnClickListener(new View.onClickListener() {
  @Override
  public boolean onClick(View v) {
    ...
    user.saveInBackground(userSaveCb);
    ...
};
```

In general, every time you catch yourself declaring a `new X() {...}` with non-trivial logic, consider extracting that into a new variable:
```
X x = new X(){...};
doSomethingWithX(x);
```